File: servlets/src/java/edu/rice/rubis/servlets/SellItemForm.java
/*
 * Copyright (C) 2002-2009  OW2 Consortium
 *
 * This file is part of dcsj-rubis (below referred to as "this program").
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Lesser General Public License as published
 * by the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

package edu.rice.rubis.servlets;

import java.io.IOException;

import javax.servlet.ServletException;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

/** 
 * Builds the html page that display the form to register a new item to sell.
 *
 * @author <a href="mailto:cecchet@rice.edu">Emmanuel Cecchet</a>
 * @author <a href="mailto:julie.marguerite@inrialpes.fr">Julie Marguerite</a>
 * @author <a href="mailto:marco.guazzone@gmail.com">Marco Guazzone</a>
 */
public class SellItemForm extends BaseRubisHttpServlet
{
	@Override
	protected void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException
	{
		this.doGet(request, response);
	}

	@Override
	protected void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException
	{
		String categoryId = null, userId = null;

		ServletPrinter sp = null;
		sp = new ServletPrinter(response, "SellItemForm");
		sp.printHTMLheader("Sell your item");

		categoryId = request.getParameter("category");
		userId = request.getParameter("user");

		sp.printFile(Config.HTMLFilesPath + "/sellItemForm.html");
		sp.printHTML("<input type=hidden name=\"userId\" value=\"" + userId + "\"> ");
		sp.printHTML("<input type=hidden name=\"categoryId\" value=\"" + categoryId + "\"> ");
		sp.printHTMLfooter();
	}
}

File: php/SellItemForm.php
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html>
  <body>
    <?php
    $scriptName = "SellItemForm.php";
    include("PHPprinter.php");
    $startTime = getMicroTime();
    
	$category = NULL;
	if (isset($_POST['category']))
	{
    	$category = $_POST['category'];
	}
	else if (isset($_GET['category']))
	{
    	$category = $_GET['category'];
	}
	else
	{
		printError($scriptName, $startTime, "SellItemForm", "You must provide a category identifier!<br>");
		exit();
	}
	$user = NULL;
	if (isset($_POST['user']))
	{
    	$user = $_POST['user'];
	}
	else if (isset($_GET['user']))
	{
    	$user = $_GET['user'];
	}
	else
	{
		printError($scriptName, $startTime, "SellItemForm", "You must provide a user identifier!<br>");
		exit();
	}

    printHTMLheader("RUBiS: Sell your item");
    include("sellItemForm.html");
    print("<input type=hidden name=\"userId\" value=\"$user\">");
    print("<input type=hidden name=\"categoryId\" value=\"$category\">");

    printHTMLfooter($scriptName, $startTime);
    ?>
  </body>
</html>
